webapp/lib/view.dart

```diff
@@ -526,8 +543,42 @@ class ActivePackagesViewPartial {
           ..classes.add('active-package-chart__description')
           ..text = chartData
       );
-    packageElement.append(packageMainContainer);
-    packageElement.append(packageChart);
+    packageElement
+      ..append(packageMainContainer)
+      ..append(packageChart)
+      ..append(
+        new SpanElement()
+          ..classes.add('active-package-menu')
+          ..onClick.listen((event) {
+            event.stopPropagation();
+            var selectedPackageDropdown = new Element.ul()
+              ..classes.addAll(['add-tag-dropdown', 'active-package-dropdown'])
+              ..append(
+                new Element.li()
+                  ..classes.add('add-tag-dropdown__item')
+                  ..text = 'Rename'
+                  ..onClick.listen((event) {
+                    packageNameElement.contentEditable = 'true';
+                    packageNameElement.focus();
+                  })
+              )
+              ..append(
+                new Element.li()
+                  ..classes.add('add-tag-dropdown__item')
+                  ..text = 'Duplicate'
+                  ..onClick.listen((event) {
+                    controller.command(controller.UIAction.duplicatePackage, controller.PackageConfigurationData(packageId, packageName));
+                  })
+              );
+            packageElement.append(selectedPackageDropdown);
+            var documentOnClickSubscription;
+            documentOnClickSubscription = document.onClick.listen((event) {
```

Comment:
I think we need to exclude from this click listener clicks on the `packageNameElement` - I think the easiest would be to add a click listener to it and just call `event.stopPropagation()`

Reply:
Implemented this though wasn't as obvious